### PR TITLE
[GOOGLEAPPS-22] added support to proxy

### PR DESCRIPTION
**Pull request by andrea-patricelli**

### CODE REVIEW

File: src/main/java/net/tirasa/connid/bundles/googleapps/GoogleAppsConfiguration.java [hunk 1]
```diff
@@ -192,6 +209,61 @@ public void setRemoveLicenseOnDisable(final boolean removeLicenseOnDisable) {
         this.removeLicenseOnDisable = removeLicenseOnDisable;
     }
 
+    @ConfigurationProperty(displayMessageKey = "proxyServerType.display",
```
Comment (ilgrosso) [inline src/main/java/net/tirasa/connid/bundles/googleapps/GoogleAppsConfiguration.java:212]: We should be using standard Java properties for HTTP proxy, not custom.

See https://docs.oracle.com/javase/6/docs/technotes/guides/net/proxies.html

File: src/main/java/net/tirasa/connid/bundles/googleapps/GoogleAppsConfiguration.java [hunk 2]
```diff
@@ -226,9 +227,34 @@ public void validate() {
...
+                                && System.getProperty("https.proxyPort") != null)
+                                ? Proxy.Type.HTTP
+                                : System.getProperty("socksProxyHost") != null
+                                        && System.getProperty("socksProxyPort") != null
+                                        ? Proxy.Type.SOCKS
+                                        : Proxy.Type.DIRECT;
+                
+                if (Proxy.Type.HTTP == proxyType) {
+                    HttpClientBuilder clientBuilder = HttpClientBuilder.create();
+                    clientBuilder.useSystemProperties();
```
Comment (ilgrosso) [inline src/main/java/net/tirasa/connid/bundles/googleapps/GoogleAppsConfiguration.java:244]: AFAIK `useSystemProperties()` will configure this builder to take the proxy configuration (among other things) from system properties, hence I think all the statement below should be safely removed.

https://www.demo2s.com/java/apache-httpclient-httpclientbuilder-tutorial-with-examples.html

**Review by ilgrosso — CHANGES_REQUESTED** [review]
See my comment

File: src/main/java/net/tirasa/connid/bundles/googleapps/GoogleAppsConfiguration.java (hunk 2 above)
Comment (andrea-patricelli) [reply src/main/java/net/tirasa/connid/bundles/googleapps/GoogleAppsConfiguration.java:244]: already tried without the further code, but still does not work without explicit proxy configuration. Going to investigate better.

Comment (ilgrosso) [reply src/main/java/net/tirasa/connid/bundles/googleapps/GoogleAppsConfiguration.java:244]: I see.

Can I suggest anyway to read the system properties just one, to improve readability? Here and in the other modified code.

Comment (andrea-patricelli) [reply src/main/java/net/tirasa/connid/bundles/googleapps/GoogleAppsConfiguration.java:244]: Sure much better! Going to do this

Comment (andrea-patricelli) [reply src/main/java/net/tirasa/connid/bundles/googleapps/GoogleAppsConfiguration.java:244]: Confirm from DEBUG logging on `org.apache.http.impl.conn` version `4.5.14` that  system properties are not taken into account. That's really strange

Comment (ilgrosso) [reply src/main/java/net/tirasa/connid/bundles/googleapps/GoogleAppsConfiguration.java:244]: Ok, so let's leave the code with explicit settings as it is

**Review by ilgrosso — APPROVED** [review]

**Pull request merged**